File: scheduler/core/src/main/java/io/orkes/conductor/scheduler/config/SchedulerProperties.java
package io.orkes.conductor.scheduler.config;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

import lombok.Getter;
import lombok.Setter;

@Configuration
@ConfigurationProperties("conductor.scheduler")
@Getter
@Setter
public class SchedulerProperties {

    private int pollingThreadCount = 1;
    private int archivalThreadCount = 2;
    private String schedulerTimeZone = "UTC";

    private int pollingInterval = 100;
    private int pollBatchSize = 5;
    private int archivalPollBatchSize = 5;

    private int archivalMaintenanceIntervalRecordCount = 5000;
    private int archivalMaintenanceLockSeconds = 600;
    private int archivalMaintenanceLockTrySeconds = 1;
    private int archivalMaxRecords = 5;
    private int archivalMaxRecordThreshold = 10;
    private int maxScheduleJitterMs = 1000;
    private long initialDelayMs = 15000;

    private int userCacheExpireAfterWriteSeconds = 120;
    private int userCacheMaxSize = 1000;

    /** When true, the scheduler uses an external {@code SchedulerCacheDAO} for hot-path lookups. */
    private boolean cacheEnabled = false;
}

File: scheduler/core/src/main/typescript/io/orkes/conductor/scheduler/config/SchedulerProperties.ts
export class SchedulerProperties {
    public pollingThreadCount: number = 1;
    public archivalThreadCount: number = 2;
    public schedulerTimeZone: string = "UTC";

    public pollingInterval: number = 100;
    public pollBatchSize: number = 5;
    public archivalPollBatchSize: number = 5;

    public archivalMaintenanceIntervalRecordCount: number = 5000;
    public archivalMaintenanceLockSeconds: number = 600;
    public archivalMaintenanceLockTrySeconds: number = 1;
    public archivalMaxRecords: number = 5;
    public archivalMaxRecordThreshold: number = 10;
    public maxScheduleJitterMs: number = 1000;
    public initialDelayMs: number = 15000;

    public userCacheExpireAfterWriteSeconds: number = 120;
    public userCacheMaxSize: number = 1000;

    /** When true, the scheduler uses an external SchedulerCacheDAO for hot-path lookups. */
    public cacheEnabled: boolean = false;
}
